docs/UserGuide.md

```diff
@@ -116,19 +116,107 @@ Examples:
 - `edit 1 p/999` will overwrite the 1st client's phone number to `999`.
 - `edit 2 n/Kylie  i/3333 f/5` will overwrite the 2nd client's name to `Kylie`, income to `3333` and family size to `5`.
 
-### Filtering clients : `filter`
-
-Filters the list of client with an input keyphrase.
-
-Format: `filter KEYPHRASE`
-
-- The search is case-insensitive. e.g `james` will match `James`
-- Partial words will still be matched e.g. `Udh` will match `Udhaya`
-- All persons' names containing the keyword will be returned e.g. `Al` will return `Alicia`, `Allysa`
-
-Example:
-
-- `filter Al` will list out persons whose name has `"Al"` inside, such as `"Alicia"`, `"Allysa"` and `"Jamal"`
+### Filtering clients: `filter`
+
+The filter command in Realodex allows you to narrow down your list of clients by specifying a keyphrase 
+related to one of the client's attributes: name, remark, tag, birthday month, or housing type.
+This feature is particularly useful when you need to focus on a
+subset of your client database that meets certain criteria.
+
+>
+> When performing multiple filter operations in sequence, 
+> each new filter is applied to the original, full list of clients, not the subset produced by the previous filter. 
+> This approach ensures clarity and consistency in search results
+
+<u>Filter by Name format:</u> `filter n/KEYPHRASE`
+
+- Returns the list of clients whose names contain the specified keyphrase.
+- Keyphrase input should be in valid format for names (alphanumeric) and non-empty.
+- The search is **case-insensitive**.
```

Comment:
Since all searches are case-insensitive, perhaps you can put it at the start tgt with the multiple filter clause, because its also a feature

Reply:
facts added new issue